pytorch_lightning/utilities/meta.py

```diff
@@ -186,7 +186,7 @@ def materialize_module(root_module: nn.Module) -> nn.Module:
         if not materialize_fn or isinstance(child, (Sequential, ModuleList, ModuleDict)):
             materialize_module(child)
         else:
-            setattr(child, name, materialize_fn())
+            setattr(root_module, name, materialize_fn())
```

Comment:
I can't think of a case when `setattr(child, name, materialize_fn())` is valid here. Is there? cc @tchaton as the main author.

---

For example, when we materialize `BoringModel`, their values are:

```
root_module=BoringModel(...)
child=Linear(...)
name="layer"
materialize_fn()=Linear(...)
```